samples/tuples/destructuring_assignment.jakt

```diff
@@ -1,11 +1,48 @@
 /// Expect:
-/// - output: "1 hello\n"
+/// - output: "1 hello\n1 hello\n2, 3, 4\n1, 2, 3\n"
 
 function foo() -> (i64, String) {
     return (1, "hello")
 }
 
+struct Test {
+    a: i64
+    b: i64
+    c: i64
+}
+
 function main() {
     let (x1, x2) = foo()
-    println("{} {}", x1, x2)
+
+    print("{} ", x1)
+    println("{}", x2)
+
+    mut bar = 0
+    mut baz = ""
+
+    (bar, baz) = foo()
+
+    print("{} ", bar)
+    println("{}", baz)
+
+    let tuple = (1, 2, 3)
+
+    mut (a, b, c) = tuple
+    a++
+    b++
+    c++
+
+    print("{}, ", a)
+    print("{}, ", b)
+    println("{}", c)
+
+    let nested_tuple = (1, (2, 3))
+
+    mut test = Test(a: 5, b: 6, c: 7)
+
+    (test.a, (test.b, test.c)) = nested_tuple
```

Comment:
IMHO destructuring should go into bindings not expressions. In the above you're overloading the concept of assignment and binding, but I think you want to keep binding a bit more pure. For example, this would be fine:

```
let (e, (f, g)) = nested_tuple
```